types/src/transaction/mod.rs

```diff
@@ -679,6 +687,8 @@ pub enum TransactionPayload {
     /// Contains an executable (script/entry function) along with extra configuration.
     /// Once this new format is fully rolled out, above payload variants will be deprecated.
     Payload(TransactionPayloadInner),
+    /// Represents an encrypted transaction payload
+    EncryptedPayload(EncryptedPayload),
```

Comment:
Why not adding it to `TransactionPayloadInner`?

Reply:
The variants were versioned with V1..., so adding `Encrypted` variant felt awkward, thats the only reason.